pilot/pkg/model/push_context.go

```diff
@@ -1558,6 +1575,41 @@ func (ps *PushContext) initMeshNetworks() {
 	}
 }
 
+func (ps *PushContext) initClusterLocalHosts() {
+	// Create the default list of cluster-local domains.
+	clusterLocalDomains := make([]string, 0, len(defaultClusterLocalNamespaces))
+	for _, n := range defaultClusterLocalNamespaces {
+		clusterLocalDomains = append(clusterLocalDomains, "."+n+".svc.cluster.local")
```

Comment:
`cluster.local` is configurable, I am not sure we can make it a constant here. I think its passed into pilot as a flag? or somehow passed in

Reply:
oof right ... the "domain" ... let me poke around to see which structure has it.

Reply:
Actually, if the user specifies a domain suffix of `mycompany.com`, will the `kube-dns` service then be referenced by `kube-dns.kube-system.mycompany.com`? That doesn't seem quite right.

Reply:
I am pretty sure? I have never done it but  Ithought that was the whole point of domain?

Reply:
Correct

Reply:
@howardjohn I had to fish through the domain suffix. The easiest place to put it was Environment ... I don't love it, but it seemed the lesser of many evils.  PTAL